doc/grammars/type.abnf

```diff
@@ -6,19 +6,26 @@ Type
 	= Atomic [Union / Intersection]
 	/ Nullable
 
+ParenthesizedType
+	= Atomic [Union / Intersection / Conditional]
+	/ Nullable
+
 Union
 	= 1*(TokenUnion Atomic)
 
 Intersection
 	= 1*(TokenIntersection Atomic)
 
+Conditional
+    = 1*ByteHorizontalWs TokenIs [TokenNot] Atomic TokenNullable Atomic TokenColon Atomic
```

Comment:
not sure how these `abnf` file are sensitive to whitespace, but the newly introduced lines use spaces, not tabs